From 4bb471d853d4a7d540db402074a3c43045c98af6 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Tue, 31 May 2016 17:32:20 +0300
Subject: [PATCH 0268/1345] pp3: Add nss0 network device by default

commit  1e949e84b36c8cbfe1b41de4739050622d678293 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- virtual port number is 31
- behavior as any nssX network device

Change-Id: I0987dbbca83ac020fee26e23bb45cfee1a8f3db5
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30223
Reviewed-by: Dovrat Zifroni <dovrat@marvell.com>
Tested-by: Dovrat Zifroni <dovrat@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
Reviewed-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../ethernet/marvell/pp3/gnss/mv_pp3_gnss_api.c    |    6 +++++-
 drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c |    5 ++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/pp3/gnss/mv_pp3_gnss_api.c b/drivers/net/ethernet/marvell/pp3/gnss/mv_pp3_gnss_api.c
index bfbd73e..ea22113 100644
--- a/drivers/net/ethernet/marvell/pp3/gnss/mv_pp3_gnss_api.c
+++ b/drivers/net/ethernet/marvell/pp3/gnss/mv_pp3_gnss_api.c
@@ -93,7 +93,11 @@ int mv_pp3_gnss_dev_create(unsigned short vport, bool state, unsigned char *mac)
 	struct pp3_dev_priv *dev_priv;
 	int msec;
 
-	sprintf(name, "nss%d", vport);
+	/* Special name nss0 for network interface used for default gateway */
+	if (vport == MV_NSS_EXT_PORT_MAX)
+		sprintf(name, "nss%d", 0);
+	else
+		sprintf(name, "nss%d", vport);
 
 	/* check MAC validation */
 	if (mac && !is_valid_ether_addr(mac))
diff --git a/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c b/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
index eed35da..dbc0abd 100644
--- a/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
+++ b/drivers/net/ethernet/marvell/pp3/platform/mv_pp3.c
@@ -48,6 +48,7 @@
 #include "net_dev/mv_dev_sysfs.h"
 #include "common/mv_sw_if.h"
 #include "a390_gic_odmi_if.h"
+#include "gnss/mv_pp3_gnss_api.h"
 
 #define MV_PP3_SHARED_NAME        "mv_pp3_shared"
 
@@ -884,7 +885,6 @@ static int mv_pp3_shared_probe(struct platform_device *pdev)
 	mv_pp3_netdev_global_init(pp3_device);
 
 	/* Create netdevice interfaces if needed */
-
 	for_each_child_of_node(np, child) {
 		if (!of_device_is_available(child))
 			continue;
@@ -900,6 +900,9 @@ static int mv_pp3_shared_probe(struct platform_device *pdev)
 			mv_pp3_netdev_show(dev);
 		}
 	}
+	if (mv_pp3_gnss_dev_create(MV_NSS_EXT_PORT_MAX, false, NULL))
+		return -ENODEV;
+
 	if (pp3_sysfs_init(pp3_device) < 0)
 		pr_err("NSS init - sysfs initialization failed\n");
 
-- 
1.7.9.5

